mulighetsrommet-api/src/main/kotlin/no/nav/mulighetsrommet/api/services/SanityTiltaksgjennomforingService.kt

```diff
@@ -0,0 +1,94 @@
+package no.nav.mulighetsrommet.api.services
+
+import io.ktor.http.*
+import kotlinx.serialization.json.decodeFromJsonElement
+import no.nav.mulighetsrommet.api.clients.sanity.SanityClient
+import no.nav.mulighetsrommet.api.domain.dto.*
+import no.nav.mulighetsrommet.api.repositories.TiltaksgjennomforingRepository
+import no.nav.mulighetsrommet.database.utils.getOrThrow
+import no.nav.mulighetsrommet.domain.dto.TiltaksgjennomforingAdminDto
+import no.nav.mulighetsrommet.serialization.json.JsonIgnoreUnknownKeys
+import org.slf4j.LoggerFactory
+import java.util.*
+
+class SanityTiltaksgjennomforingService(
+    private val sanityClient: SanityClient,
+    private val tiltaksgjennomforingRepository: TiltaksgjennomforingRepository,
+) {
+    private val log = LoggerFactory.getLogger(javaClass)
+
+    suspend fun syncTiltaksgjennomforingerTilSanity(dryRun: Boolean) {
+        val gjennomforingerSomMangler = tiltaksgjennomforingRepository.getTiltaksgjennomforingerUtenSanityId()
+            .getOrThrow()
+        val opprettet = gjennomforingerSomMangler
+            .sumOf { opprettSanityTiltaksgjennomforing(it, dryRun) }
+        log.info(
+            "Lagde $opprettet Sanity dokumenter for ${gjennomforingerSomMangler.size}" +
+                " tiltaksgjennomforinger som manglet sanity_id",
+        )
+    }
+
+    private suspend fun oppdaterIdOmAlleredeFinnes(tiltaksgjennomforing: TiltaksgjennomforingAdminDto): Boolean {
+        val tiltaksnummer = tiltaksgjennomforing.tiltaksnummer ?: return false
+
+        val sanityTiltaksgjennomforinger = hentTiltaksgjennomforinger(tiltaksnummer)
+        if (sanityTiltaksgjennomforinger.size > 1) {
+            throw Exception("Fant ${sanityTiltaksgjennomforinger.size} sanity dokumenter med tiltaksnummer: $tiltaksnummer")
+        }
+        if (sanityTiltaksgjennomforinger.isEmpty()) {
+            return false
+        }
+        tiltaksgjennomforingRepository.updateSanityTiltaksgjennomforingId(
+            tiltaksgjennomforing.id,
+            sanityTiltaksgjennomforinger[0]._id,
+        ).getOrThrow()
+        return true
+    }
+
+    suspend fun opprettSanityTiltaksgjennomforing(
+        tiltaksgjennomforing: TiltaksgjennomforingAdminDto,
+        dryRun: Boolean = false,
+    ): Int {
+        if (oppdaterIdOmAlleredeFinnes(tiltaksgjennomforing)) {
+            return 0
+        }
+
+        val sanityTiltaksgjennomforingId = UUID.randomUUID().toString()
+        val sanityTiltaksgjennomforing = SanityTiltaksgjennomforing(
+            _id = sanityTiltaksgjennomforingId,
+            tiltaksgjennomforingNavn = tiltaksgjennomforing.navn,
```

Comment:
Her kan man synce flere felter, men det kan være en egen oppgave kanskje. Testa litt og for og f. eks sette enhet trenger man den _ref eller _key verdien fra Sanity. Der kunne vi vel bare spurt med query kall og slengt på en cache.